dart/constraint/ConstraintSolver.cpp

```diff
@@ -441,7 +441,11 @@ DART_SUPPRESS_DEPRECATED_END
       mSoftContactConstraints.push_back(
             std::make_shared<SoftContactConstraint>(contact, mTimeStep));
     }
-    else
+    // If penetration depth is negative, then the collision isn't really
+    // happening and the contact point should be ignored.
+    // TODO(MXG): Investigate ways to leverage the proximity information of a
+    //            negative penetration to improve collision handling.
+    else if(contact.penetrationDepth >= 0.0)
```

Comment:
I think we would like to ignore negative penetration contacts even for soft contact liek:
```cpp
if(contact.penetrationDepth < 0.0)
  continue;

if (isSoftContact(contact))
{
  ...
}
else
{
  ...
}
```

Reply:
@mxgrey Any thoughts on this? :smile:

Reply:
Sorry for the delay, I got sidetracked. Makes sense to me! I've updated the branch with this change.

Reply:
Thanks! Let's merge once CI tests are done.